packages/astro/src/core/dev/index.ts

```diff
@@ -274,7 +274,7 @@ export class AstroDevServer {
   private async handleRequest(req: http.IncomingMessage, res: http.ServerResponse, next: NextFunction) {
     if (!this.viteServer) throw new Error(`AstroDevServer.start() not called`);
 
-    let pathname = req.url || '/'; // original request
+    let [pathname, search = undefined] = (req.url || '/').split('?'); // original request
```

Comment:
`= undefined` technically unnecessary but it helps TypeScript infer the correct type